client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -21,4 +23,45 @@ class Commit(
     case other: Commit => commit.commit_sha.get == other.commit.commit_sha.get
     case _ => false
   }
+
+  /** Creates a branch at this Commit and returns the checked-out branch
+   *  If the branch already exists, it will be moved to this commit.
+   *  @param branch branch name
+   *  @return if not saved, a failure; otherwise, this commit as the head of `branch`
+   */
+  def newBranch(branch: String)(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Failure(new IllegalStateException("Commit must be saved before it can be attached to a branch"))
+    else setBranch(branch)
```

Comment:
@convoliution this seems different from the client, right? We decided to return a new commit or something like that since overwriting was confusing.

Reply:
I think you're right actually. Will fix this, and reflect in the tests as well.

Reply:
Resolved. I make a request to set branch, then use repository to get that branch (analogously to the Python client).